main.go

```diff
@@ -0,0 +1,209 @@
+// parameterName = "Parent.child1.child2. ... .childn"
+// [parameterName-1 value-1, ...., parameterName-n value-n ] -> [[parent-1, child-1, ...child-n, value1],...]
+func parseArgs(args []string) [][]string {
+	var splitParameters [][]string
+	argsLength := len(args)
+	for index, parameterNames := range args[1:] {
+		if index%2 == 0 || index+2 == argsLength {
+			continue
+		}
+		parameterAndValue := strings.Split(strings.TrimPrefix(parameterNames, "--"), ".")
+		parameterAndValue = append(parameterAndValue, args[index+2])
+		splitParameters = append(splitParameters, parameterAndValue)
+	}
+	return splitParameters
+}
+
+// Iterates over all parameters and tries to set them
+func decorateType(serviceType runtime.Object, fieldNames [][]string) (runtime.Object, error) {
+	for _, v := range fieldNames {
+		_, err := setParameter(serviceType, v[:len(v)-1], v[len(v)-1])
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return serviceType, nil
+}
+
+// Iterates through service Type to find and set the required parameters
+// logs error and exits if parameter does not match any in Type specified field names
+func setParameter(serviceType runtime.Object, parameterHierarchy []string, value string) (runtime.Object, error) {
+	reflectedServiceType := reflect.ValueOf(serviceType).Elem()
+
+	for _, parameterName := range parameterHierarchy {
+		parameterName = strcase.ToCamel(parameterName)
+
+		if !reflectedServiceType.FieldByName(parameterName).IsValid() {
+			originalParameterName := parameterName
+			var err error
+			parameterName, err = fixStringCase(structs.Names(reflectedServiceType.Interface()), parameterName)
+			if err != nil {
+				err = fmt.Errorf("%w\n%s contains field with name %s : %t",
+					err,
+					reflectedServiceType.Type().Name(),
+					parameterName,
+					reflectedServiceType.FieldByName(parameterName).IsValid(),
+				)
+				return nil, err
+			}
+			info := fmt.Sprintf(
+				"Converted input parameter %s to %s to match correct fieldName\n",
+				originalParameterName,
+				parameterName,
+			)
+			logrus.Info(info)
+		}
+
+		if reflectedServiceType.FieldByName(parameterName).Kind() == reflect.Struct {
+			reflectedServiceType = reflectedServiceType.FieldByName(parameterName)
+		} else if reflectedServiceType.FieldByName(parameterName).Kind() == reflect.String {
+			reflectedServiceType = reflectedServiceType.FieldByName(parameterName)
+			reflectedServiceType.SetString(value)
+			fmt.Printf("setting\n-----------------------\nfield: %s\nvalue: %s\n-----------------------\n", strings.Join(parameterHierarchy, "."), value)
+		} else if reflectedServiceType.FieldByName(parameterName).Kind() == reflect.Int {
+			reflectedServiceType = reflectedServiceType.FieldByName(parameterName)
+			intValue, _ := strconv.Atoi(value)
+			reflectedServiceType.SetInt(int64(intValue))
+			fmt.Printf("setting\n-----------------------\nfield: %s\nvalue: %s\n-----------------------\n", strings.Join(parameterHierarchy, "."), value)
+		} else {
```

Comment:
We should probably also support float and double.

Reply:
See review comment below.